### PR TITLE
Flatten the synapse.rest.client package

### PR DESCRIPTION
**Pull request by reivilibre**
Because the `v1` and `v2_alpha` subpackages are downright LIES.

This will probably cause a couple of conflicts in the import lines, but otherwise should be fairly conflict-free (because I structured it as a `git mv` (pure rename) to start with, before going back and doing the move-with-updated-references).

This should be a commit-by-commit-reviewable PR (since each commit tries to be one mechanical step)

### CODE REVIEW

File: synapse/rest/client/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2015, 2016 OpenMarket Ltd
```
Comment (clokep) [inline synapse/rest/client/__init__.py:1]: Looks like we gained a year?

Anyway this should get updated to the m.org foundation.

**Review by clokep — APPROVED** [review]
LGTM if CI passes.

File: synapse/rest/client/__init__.py (hunk 1 above)
Comment (reivilibre) [reply synapse/rest/client/__init__.py:1]: the __init__ in the subpackages said 2014-2016, so I widened it.
Should this be updated? I don't know the legal interpretation of that, but I thought we kept that around and M.org came in after (so the old copyright still exists).

Not that any of this makes much sense on an empty file, though...

Comment (clokep) [reply synapse/rest/client/__init__.py:1]: We update them to the matrix.org foundation for openmarket and new vector.

**Comment by reivilibre** [comment]
Uh oh, a juicy conflict popped up (someone created `v2_alpha/room.py` which conflicts with `v1/room.py` as far as renames go.

**Comment by clokep** [comment]
> Should I merge the files together or leave the v2 one renamed as `roomv2` (this is the name it was imported with anyway)?

`synapse/rest/client/v1/room.py` is already very large (~1.1k LOC), I think renaming `synapse/rest/client/v2_alpha/room.py` makes the most sense since that file is large as well.  (Maybe something like `room_batch.py` or something. @MadLittleMods might have ideas.)

File: synapse/rest/__init__.py [hunk 2]
```diff
@@ -14,49 +14,47 @@
...
+    presence,
+    profile,
+    push_rule,
+    pusher,
     read_marker,
     receipts,
     register,
     relations,
     report_event,
-    room as roomv2,
```
Comment (MadLittleMods) [inline synapse/rest/__init__.py:50]: For reference, this change came from https://github.com/matrix-org/synapse/pull/10576

**Comment by MadLittleMods** [comment]
> I think renaming synapse/rest/client/v2_alpha/room.py makes the most sense since that file is large as well. (Maybe something like room_batch.py or something.
>
> -- https://github.com/matrix-org/synapse/pull/10600#issuecomment-899460109

Either `room_historical_batch.py` or `room_batch.py` sound good to me 👍

**Review by clokep — APPROVED** [review]
LGTM if CI passes.

**Pull request merged**